paimon-core/src/main/java/org/apache/paimon/catalog/CachingCatalog.java

```diff
@@ -227,26 +225,17 @@ public Table getTable(Identifier identifier) throws TableNotExistException {
             return table;
         }
 
-        if (isSpecifiedSystemTable(identifier)) {
-            Identifier originIdentifier =
-                    new Identifier(
-                            identifier.getDatabaseName(),
-                            identifier.getTableName(),
-                            identifier.getBranchName(),
-                            null);
-            Table originTable = tableCache.getIfPresent(originIdentifier);
-            if (originTable == null) {
-                originTable = wrapped.getTable(originIdentifier);
-                putTableCache(originIdentifier, originTable);
-            }
+        // For system table, do not cache it directly. Instead, cache the origin table and then wrap
+        // it to generate the system table.
+        if (identifier.isSystemTable()) {
+            Table originTable = getTable(identifier.toOriginTable());
```

Comment:
Just keep:
```
Identifier originIdentifier =
                    new Identifier(
                            identifier.getDatabaseName(),
                            identifier.getTableName(),
                            identifier.getBranchName(),
                            null);
```

It is hard to understand `toOriginTable`, with branch, or not with branch?